### PR TITLE
added competitive programming section under Youtube

### LINKED ISSUE #650: [Add] <Competitive Programming section under Youtube>
### Site URL

https://www.youtube.com/@Errichto

### Description of the above provided site

- https://www.youtube.com/@Errichto
- https://www.youtube.com/channel/UCGS5ZzcSAymQbWZvNoKOFhQ
- https://www.youtube.com/c/PriyanshAgarwal

These channels are very helpful for a person who is into competitive programming.

### Category

Youtube

### Sub Category

computer science 

### Checklist

- [X] I have checked the existing issues
- [X] I have read the [Contributing Guidelines](https://github.com/rupali-codes/LinksHub/blob/main/CONTRIBUTING.md)
- [X] I am willing to work on this issue (optional)
- [X] I am a GSSoC'23 contributor

### PR DESCRIPTION
**Pull request by harshit-002**
## Fixes Issue 
Closes #650 

## Changes proposed

- added a competitive programming section under Youtube
- added 3 channels under that category.

## Screenshots
Screenshot 2023-05-23 193714

### CODE REVIEW

File: database/youtube/competitive-programming.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
+[{
+    "name": "Errichto",
+    "description": "This channel provides competitive programming tactics and thought process for advanced alogrithms .",
```
Comment (CBID2) [inline database/youtube/competitive-programming.json:3]: [Typo]
```suggestion
    "description": "This channel provides competitive programming tactics and thought process for advanced algorithms.",
```

File: database/youtube/competitive-programming.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    "url": "https://www.youtube.com/@Errichto",
+    "category": "youtube",
+    "subcategory": "competitive-programming",
+    "language": "english"
+},
+{
+    "name": "Priyansh Agarwal",
+    "description": "This channel provides free video solutions of coding contests of codeforces, codechef etc.",
```
Comment (CBID2) [inline database/youtube/competitive-programming.json:11]: [Proofreading] This description is a bit unclear. What do you mean by “ free video solutions of coding contests of codeforces, codechef, etc“?
```suggestion
    "description": "This channel provides free video solutions of coding contests of codeforces, codechef, etc."
```

**Review by CBID2 — CHANGES_REQUESTED** [review]
Hi @harshit-002! :) I just provided some suggestions to improve your descriptions. In the future, I highly recommend running them through Grammarly or spellcheck before you submit your PR.

**Comment by harshit-002** [comment]
> Hi @harshit-002! :) I just provided some suggestions to improve your descriptions. In the future, I highly recommend running them through Grammarly or spellcheck before you submit your PR.

Sorry about that, I will make sure to double-check spellings.

**Review by harshit-002 — COMMENTED** [review]
Removed typos and made the descriptions clear.

**Review by CBID2 — APPROVED** [review]
LGTM! 😊

**Review by rupali-codes — APPROVED** [review]
looks good

**Pull request merged**